### ISSUE TITLE
History (chatlog) usability

### ISSUE THREAD

**Issue by redmanmale** [uTox/uTox#1089]
Here's two points.

1. **Load more (maybe all) message from log**. Time to time I need to search through history for more than 256 (UTOX_MAX_BACKLOG_MESSAGES) messages away from now. It would be great to make this configurable option (in UI settings section).
What can stop us from doing so?

2. **Search through history**. Right now the only option is to export chat log and grep through. It's not very usable. It would be great to make search panel; like in Skype.
![pic](https://puu.sh/xVv8M/c179fa9321.png)

**Comment by GrayHatter** [comment]
@redmanmale nothing is stopping us.

That's why I wrote the chatlog functions the way I did. You can call into get_chatlog with a skip number, and it'll load that number of lines, starting skip number from the last. (I may not be correct on the order/direction, I'm trying to answer from memory). 

The problem is managing the memory is involved. My original thought, was to stripout the realloc section of messages.c instead using a double linked list for messages, with friend->msg pointing to the newest message, with friend->msg->older, and friend->msg->newer. doing it that way makes it much easier to load a long history messages, free them, and search through disk history. At the expense of longer load times when uTox draws these messages at any other position than the bottom.

I haven't made the time to fix this, but it shouldn't be too difficult a project.

@robinlinden @endoffile78 @cebe thoughts?

**Comment by cebe** [comment]
yeah, memory management is the main issue here as far as I see. If we have a way to dynamically load messages it should be quite easy to implement.
